### DISCUSSION TITLE
precedence and security checks in RFC793

### DISCUSSION THREAD

**Discussion by livelykitten** [ANLAB-KAIST/KENSv3#145] [Project Q&A 2. Transport Layer (TCP)] [answered]
RFC793 requires us to implement precedence level in TCP, but I heard that this feature was rarely used even before RFC793 became obsolete.

Do we need to implement precedence?

**Comment by Pusnow** [comment, marked as answer]
You don't need to implement all TCP features (including precedence) in RFCs as long as you pass all test cases.
